bundles/org.openhab.binding.androiddebugbridge/src/main/java/org/openhab/binding/androiddebugbridge/internal/AndroidDebugBridgeDevice.java

```diff
@@ -202,6 +202,10 @@ public void stopPackage(String packageName)
     public String getCurrentPackage() throws AndroidDebugBridgeDeviceException, InterruptedException,
             AndroidDebugBridgeDeviceReadException, TimeoutException, ExecutionException {
         var out = runAdbShell("dumpsys", "window", "windows", "|", "grep", "mFocusedApp");
+        if (getAndroidVersion().contains("11")) {
+            out = runAdbShell("dumpsys", "window", "displays", "|", "grep", "mFocusedApp");
```

Comment:
Is is on purpose that you run both commands for Android 11, i.e. line 204 **and** 206?

Reply:
Honestly i'm not formally trained on coding or anything and I was having issues. My original idea was just to make a if else statement that depending on the string for getAndroidVersion would either declare the out variable as either line 204 or 206 depending. but when I made it like this it would fail to compile saying out was not defined

if (getAndroidVersion().contains("11")) {
var out = runAdbShell("dumpsys", "window", "displays", "|", "grep", "mFocusedApp");
} else {
var out = runAdbShell("dumpsys", "window", "windows", "|", "grep", "mFocusedApp");
}

and I don't understand the concept of classes and public vs private etc etc. I get a surprising amount of things done considering but a lot of trial and error.

Reply:
```
String out;
if (getAndroidVersion().contains("11")) {
    out = runAdbShell("dumpsys", "window", "displays", "|", "grep", "mFocusedApp");
} else {
    out = runAdbShell("dumpsys", "window", "windows", "|", "grep", "mFocusedApp");
}
```

or perhaps:

```
String out = runAdbShell("dumpsys", "window", getAndroidVersion().contains("11") ? "displays" : "windows", "|", "grep", "mFocusedApp");
```

Reply:
oh I see I had to declare out before the ifdef... gotcha.. but your suggestion seems better. I"ll update it and change it to >= 11